chapters/arrays.tex

```diff
@@ -1268,10 +1268,20 @@ \subsection{Element-wise Division}\label{array-element-wise-division}\label{elem
 
 \subsection{Element-wise Exponentiation}\label{element-wise-exponentiation}
 
-Exponentiation \lstinline!a ^ b! is defined as \lstinline[language=C]!pow(double a, double b)! in the ANSI~C library if both \lstinline!a! and \lstinline!b! are
-\lstinline!Real! scalars. A \lstinline!Real! scalar value is returned.  If \lstinline!a! or \lstinline!b! are \lstinline!Integer! scalars, they are
-automatically promoted to \lstinline!Real!.  Consequences of exceptional situations, such as ($\text{\lstinline!a!} = 0.0$ and $\text{\lstinline!b!} \leq 0.0$,
-$\text{\lstinline!a!} < 0$ and \lstinline!b! is not an integer) or overflow are undefined.
+Exponentiation \lstinline!a ^ b! always returns a \lstinline!Real! scalar value, and it is required that \lstinline!a! and \lstinline!b! are scalar \lstinline!Real! or \lstinline!Integer! values.
+The result should correspond to mathematical exponentiation (ideally the correctly rounded result based on the inputs) with the following special cases:
+\begin{itemize}
+\item If $\text{\lstinline!a!} = 0.0$ and $\text{\lstinline!b!} = 0$ for an \lstinline!Integer! expression \lstinline!b! the result is $1.0$.
+\item If $\text{\lstinline!a!} < 0$ and \lstinline!b! is an \lstinline!Integer!, the result is defined as $\pm |a|^b$, with sign depending on whether \lstinline!b! is even (positive) or odd (negative).
+\item If $\text{\lstinline!a!} < 0$ and \lstinline!b! is a \lstinline!Real! having a non-zero integer value, the result is defined as $\pm |a|^b$, with sign depending on whether \lstinline!b! is even (positive) or odd (negative). This case is deprecated, and tools may warn once during a simulation if this occurs.
+\item Consequences of other exceptional situations, such as ($\text{\lstinline!a!} = 0.0$ and $\text{\lstinline!b!} = 0.0$ for a \lstinline!Real b!, $\text{\lstinline!a!} = 0.0$ and $\text{\lstinline!b!} < 0$, or $\text{\lstinline!a!} < 0$ and \lstinline!b! does not have an integer value) or overflow are undefined.
+\end{itemize}
+
+\begin{nonnormative}
+Except for defining the special case of $0.0^0$ it corresponds to \lstinline[language=C]!pow(double a, double b)! in the ANSI~C library.
+The result is always \lstinline!Real! due to the potential for integer overflow and negative exponents.
```

Comment:
While negative exponents is a very strong reason, I don't think that risk of overflow is a good one (after all, multiplying two `Integer` values still gives an `Integer` result, despite the risk of overflow):
```suggestion
The result is always \lstinline!Real! as negative exponents can give non-integer results also when both operands are \lstinline!Integer!.
```

Reply:
Ok. To me the risk of overflow is larger for exponentiation since the result grows so much faster, but on the other hand there's also a risk of real overflow.